docs/_include/variable_hydro.rst

```diff
@@ -0,0 +1,113 @@
+dictate the varying hydrodynamics. The total force triggers a custom PTO
+force, and the heave position determines what BEM dataset is used as the body
+submerges.
+
+User defined logic dictates how quickly the operational depth changes, and 
+selects the corresponding BEM data at the new state.
+In this case, the user could supply BEM datasets of the device at 
+30 different depths over the 3m range. User defined logic then selects 
+the BEM dataset most closely corresponding to the instantaneous 
+heave position of the device. That new BEM dataset is then used to 
+calculate the hydrodynamic forcing on the device until updated again.
+
+Implementation
+""""""""""""""
+Variable hydrodynamics is implemented by allowing users to initialize a body
+with a cell array of h5 files, as many as required. Each i-th h5 file is then pre-processed into
+``body.hydroData(i)`` and ``body.hydroForce.hf{i}`` in the normal manner.
+The entirety of body.hydroForce is loaded into Simulink using a custom bus. 
+The index ``body1_hydroForceIndex`` in Simulink, which indexes 
+``body.hydroForce.hf{i}`` and selects the hydrodynamic force coefficients used
+in the hydrodynamic force calculations at a given time.
+Users *must* create their own logic and index signal in Simulink. A ``GoTo`` 
+block named ``body1_hydroForceIndex``, ``body2_hydroForceIndex`` must take 
+the signal so that the body block can read the value and select the correct 
+hydrodynamic coefficients.
+
+Variable hydrodynamics is body dependent and does not need to be applied to 
+every body in a simulation.
+To turn on variable hydrodynamics, use the ``body.variableHydro.option`` flag in the 
+wecSimInputFile and input multiple H5 files during the body initialization::
+
+    body(1) = bodyClass({'H5FILE_1.h5','H5FILE_2.h5','H5FILE_3.h5','H5FILE_4.h5','H5FILE_5.h5');
+    body(1).variableHydro.option = 1;
+
+
+If ``body.variableHydro.option=0`` any extra H5 files in the body initialization 
+are ignored. If only one H5 file is submitted to a file, variable hydrodynamics 
+are turned off.
+
+The flag ``body.variableHydro.hydroForceIndexInitial`` allows the user to set the
+default index used to define the hydrodynamic coefficients. This initial dataset
+should represent the body at the start of a simulation and likely its equilibrium 
+position. The initial dataset will be used to define the mass, center of gravity, 
+and center of buoyancy for a simulation.
+
+This parameter is flexible because an initial index of zero is not always convenient
+and can complicate indexing logic. For example, consider a flap-based device with
+multiple hydrodynamic datasets at various pitch angles (-50:10:50). It is most convenient
+to treat these angles in numerical order in BEM simulations, indexing logic, and other 
+data processing. In this case the initial position is at a pitch angle of 0 so ``body.variableHydro.hydroForceIndexInitial=6;``.
+
```

Comment:
This level of detail seems more appropriate for the developer documentation. I recommend simplifying the user docs to only what the user needs to know to implement the feature, which is that they need to specify which index to start with. I don't even know how critical it is for a user to understand the bus signals.

Reply:
Here is the ChatGPT recommendation for Implementation

## Implementation
To implement Variable Hydrodynamics, follow these steps:

1. **Initialize the Body**: Create a body instance with a cell array of H5 files containing the hydrodynamic data. For example:

   `body(1) = bodyClass({'H5FILE_1.h5', 'H5FILE_2.h5', 'H5FILE_3.h5', ...});`

2. **Enable Variable Hydrodynamics**: Set the variableHydro.option flag to enable the feature:

   `body(1).variableHydro.option = 1;`

3. **Set Initial Hydrodynamic Coefficient Index**: Use the hydroForceIndexInitial flag to define the default index for hydrodynamic coefficients, typically representing the body’s starting condition:

   `body(1).variableHydro.hydroForceIndexInitial = <initial_index>;`

4. **Simulink Integration**: Load the hydrodynamic force data into Simulink using a custom bus. Create a GoTo block named body1_hydroForceIndex (or similar) to allow the body block to read the index signal for selecting the appropriate hydrodynamic coefficients.

5. **User-Defined Logic**: Implement your own logic in Simulink to determine when to switch between H5 files based on the specified conditions.

_Note: Variable hydrodynamics can be applied to individual bodies in a simulation, and it is not necessary for all bodies._

Reply:
I like the suggested numbering of steps for a user's implementation. I combined this with the detailed paragraphs to walk a user through set-up.